aws/src/main/java/org/apache/iceberg/aws/AwsProperties.java

```diff
@@ -493,53 +492,7 @@ public class AwsProperties implements Serializable {
   private String dynamoDbEndpoint;
 
   public AwsProperties() {
-    this.httpClientType = HTTP_CLIENT_TYPE_DEFAULT;
-    this.stsClientAssumeRoleTags = Sets.newHashSet();
-
-    this.clientAssumeRoleArn = null;
-    this.clientAssumeRoleTimeoutSec = CLIENT_ASSUME_ROLE_TIMEOUT_SEC_DEFAULT;
-    this.clientAssumeRoleExternalId = null;
-    this.clientAssumeRoleRegion = null;
-    this.clientAssumeRoleSessionName = null;
-
-    this.s3FileIoSseType = S3FILEIO_SSE_TYPE_NONE;
-    this.s3FileIoSseKey = null;
-    this.s3FileIoSseMd5 = null;
-    this.s3AccessKeyId = null;
-    this.s3SecretAccessKey = null;
-    this.s3SessionToken = null;
-    this.s3FileIoAcl = null;
-    this.s3Endpoint = null;
-
-    this.s3FileIoMultipartUploadThreads = Runtime.getRuntime().availableProcessors();
-    this.s3FileIoMultiPartSize = S3FILEIO_MULTIPART_SIZE_DEFAULT;
-    this.s3FileIoMultipartThresholdFactor = S3FILEIO_MULTIPART_THRESHOLD_FACTOR_DEFAULT;
-    this.s3FileIoDeleteBatchSize = S3FILEIO_DELETE_BATCH_SIZE_DEFAULT;
-    this.s3fileIoStagingDirectory = System.getProperty("java.io.tmpdir");
-    this.isS3ChecksumEnabled = S3_CHECKSUM_ENABLED_DEFAULT;
-    this.s3WriteTags = Sets.newHashSet();
-    this.s3DeleteTags = Sets.newHashSet();
-    this.s3FileIoDeleteThreads = Runtime.getRuntime().availableProcessors();
-    this.isS3DeleteEnabled = S3_DELETE_ENABLED_DEFAULT;
-    this.s3BucketToAccessPointMapping = ImmutableMap.of();
```

Comment:
Can we just change this line instead of replacing whole CTOR?  If I understand, its the one causing issue?

 I realize its same thing but it makes a bigger code change and I guess this version was supposed to be optimized to avoid PropertyUtil calls?

Reply:
sounds good, I'll rebase and minimize the changes.